receiver/vcenterreceiver/metrics.go

```diff
@@ -406,3 +406,57 @@ func (v *vcenterMetricScraper) recordVMPerformanceMetrics(entityMetric *performa
 		}
 	}
 }
+
+// recordVMVSANMetrics records vSAN metrics for a vSphere Virtual Machine
+func (v *vcenterMetricScraper) recordVMVSANMetrics(vSANMetrics *VSANMetricResults) {
+	for _, metric := range vSANMetrics.MetricDetails {
+		for i, value := range metric.Values {
+			timestamp := metric.Timestamps[i]
+			switch metric.MetricLabel {
+			case "iopsRead":
+				v.mb.RecordVcenterVMVsanOperationsDataPoint(pcommon.NewTimestampFromTime(*timestamp), value, metadata.AttributeVsanOperationTypeRead)
+			case "iopsWrite":
+				v.mb.RecordVcenterVMVsanOperationsDataPoint(pcommon.NewTimestampFromTime(*timestamp), value, metadata.AttributeVsanOperationTypeWrite)
+			case "throughputRead":
+				readRate := float64(value) / float64(metric.Interval)
+				v.mb.RecordVcenterVMVsanThroughputDataPoint(pcommon.NewTimestampFromTime(*timestamp), readRate, metadata.AttributeVsanThroughputDirectionRead)
+			case "throughputWrite":
+				writeRate := float64(value) / float64(metric.Interval)
+				v.mb.RecordVcenterVMVsanThroughputDataPoint(pcommon.NewTimestampFromTime(*timestamp), writeRate, metadata.AttributeVsanThroughputDirectionWrite)
+			case "latencyRead":
+				v.mb.RecordVcenterVMVsanLatencyAvgDataPoint(pcommon.NewTimestampFromTime(*timestamp), value, metadata.AttributeVsanLatencyTypeRead)
+			case "latencyWrite":
+				v.mb.RecordVcenterVMVsanLatencyAvgDataPoint(pcommon.NewTimestampFromTime(*timestamp), value, metadata.AttributeVsanLatencyTypeWrite)
+			}
+		}
+	}
+}
+
+// recordHostVSANMetrics records vSAN metrics for a vSphere host 
+func (v *vcenterMetricScraper) recordHostVSANMetrics(vSANMetrics *VSANMetricResults) {
+	for _, metric := range vSANMetrics.MetricDetails {
+		for i, value := range metric.Values {
+			timestamp := metric.Timestamps[i]
+			switch metric.MetricLabel {
+			case "iopsRead":
+				v.mb.RecordVcenterHostVsanOperationsDataPoint(pcommon.NewTimestampFromTime(*timestamp), value, metadata.AttributeVsanOperationTypeRead)
+			case "iopsWrite":
+				v.mb.RecordVcenterHostVsanOperationsDataPoint(pcommon.NewTimestampFromTime(*timestamp), value, metadata.AttributeVsanOperationTypeWrite)
+			case "throughputRead":
+				readRate := float64(value) / 20
```

Comment:
we're now pulling the metric interval so you'll want to divide by that just like in the VMs

Reply:
resolve